estate/models/estate_property.py

```diff
@@ -0,0 +1,153 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+from datetime import date
+from dateutil.relativedelta import relativedelta
+from odoo.tools.float_utils import float_compare, float_is_zero
+from odoo.exceptions import ValidationError
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+
+    _sql_constraints = [
+        (
+            "expected_price_strictly_positive",
+            "CHECK(expected_price > 0)",
+            "Expected price must be strictly positive",
+        ),
+        (
+            "selling_price_positive",
+            "CHECK(selling_price >= 0)",
+            "Selling price must be positive",
+        ),
+    ]
+
+    @api.constrains("expected_price", "selling_price")
+    def _check_selling_price(self):
+        for record in self:
```

Comment:
`record` is a bit too generic, it's fine for small method like this. But when you'll work in huuuge methods, you'll be happy to have a name a bit more meaningful trust me
```suggestion
        for estate in self:
```